### PR TITLE
Deprecate Bokeh.$ and Bokeh._

### LINKED ISSUE #5532: Deprecate Bokeh.$  and Bokeh._
This is required in preparation to remove JQuery and underscore (or replace which something smaller).

### PR DESCRIPTION
**Pull request by mattpap**
fixes #5532

### CODE REVIEW

File: bokehjs/src/coffee/main.coffee [hunk 1]
```diff
@@ -4,7 +4,16 @@ export {version} from "./version"
...
+Object.defineProperty(module.exports, "_", {
+  get: () ->
+    console.warn("Bokeh._ is deprecated. You have to provide your own copy of underscore if necessary.")
+    return _
+})
+Object.defineProperty(module.exports, "$", {
+  get: () ->
+    console.warn("Bokeh.$ is deprecated. You have to provide your own copy of jquery if necessary.")
+    return $
+})
```
Comment (bryevdv) [inline bokehjs/src/coffee/main.coffee]: Suggest "is deprecated and will be removed in the future" just to be crystal clear.

File: bokeh/embed.py [hunk 2]
```diff
@@ -30,23 +30,25 @@
...
-    return _wrap('document.addEventListener("DOMContentLoaded", function(event) {\n', code, '\n});')
+    return """\
+(function() {
+  var fn = function() {
+%(code)s
+  };
+  if (document.readyState != "loading") fn();
+  else document.addEventListener("DOMContentLoaded", fn);
+})();
+""" % dict(code=_indent(code, 4))
```
Comment (bryevdv) [inline bokeh/embed.py]: why different indent levels 4 here and 2 above?

**Comment by bryevdv** [comment]
Moved `MIGRATION` to associated issue.

File: bokeh/embed.py (hunk 2 above)
Comment (mattpap) [inline bokeh/embed.py]: Indentation in bokehjs consists of two spaces. Here it was all four and in other files mixed 2 and 4.

**Pull request merged**